datafusion/optimizer/src/optimizer.rs

```diff
@@ -240,7 +239,6 @@ impl Optimizer {
             // Filters can't be pushed down past Limits, we should do PushDownFilter after PushDownLimit
             Arc::new(PushDownLimit::new()),
             Arc::new(PushDownFilter::new()),
-            Arc::new(SingleDistinctToGroupBy::new()),
```

Comment:
Queries like `avg(distinct a)` rely on this rule, and without it, they cannot be executed anymore.

The following query is available on the main branch, but not on this PR branch.
```sh
DataFusion CLI v46.0.0
> select avg(distinct a) from values(1) t(a);
Execution error: avg(DISTINCT) aggregations are not available
```

Reply:
The alternative would be to directly implement distinct support in the avg accumulator

The lack of avg support means we can't run multiple distinct avg aggregates

```sql
> select avg(distinct a), avg(distinct b) from values(1, 2) t(a, b);
```

This is tracked by 
- v

Reply:
I also updated the ticket https://github.com/apache/datafusion/issues/2408 to explain the issue and a bit more about what is needed